### PR TITLE
chore: swap in libp2p rpc module

### PR DESCRIPTION
**Pull request by sanderpick**
Nothing wild, just replacing `pubsub` with https://github.com/textileio/go-libp2p-pubsub-rpc. I also moved `marketpeer` and `finalizer` there. Finalizer was started by Anton from Anytype in `go-threads`. Ideally he can create a repo for that and we could use it... I asked him if he wanted to do that.

### CODE REVIEW

**Comment by sanderpick** [comment]
Fixed some import collisions related to the latest `ipfs-lite` and the filecoin tooling. Downgrading `go-ipns` did the trick.

File: go.mod [hunk 1]
```diff
@@ -62,6 +55,4 @@ require (
 
 replace github.com/kilic/bls12-381 => github.com/kilic/bls12-381 v0.0.0-20200820230200-6b2c19996391
 
-replace github.com/ethereum/go-ethereum => github.com/textileio/go-ethereum v1.10.3-0.20210413172519-62e8b38d82b1
```
Comment (sanderpick) [inline go.mod:65]: I dropped this replacement, but let me know if this was really needed for something.

Comment (merlinran) [reply go.mod:65]: No these are probably leftovers when extracting this out.

**Review by merlinran — APPROVED** [review]
LGTM! Cool to see that piece of work lives on the separate repo.

**Pull request merged**